atlasdb-client/src/main/java/com/palantir/atlasdb/transaction/impl/CachingTransaction.java

```diff
@@ -124,7 +124,28 @@ public Map<Cell, byte[]> get(TableReference tableRef, Set<Cell> cells) {
             return getWithLoader(
                             tableRef,
                             cells,
-                            (tableReference, toRead) -> Futures.immediateFuture(super.get(tableReference, toRead)))
+                            (tableReference, _cachedCells, toRead) ->
+                                    Futures.immediateFuture(super.get(tableReference, toRead)))
+                    .get();
+        } catch (InterruptedException | ExecutionException e) {
+            throw Throwables.rewrapAndThrowUncheckedException(e.getCause());
+        }
+    }
+
+    @Override
+    public Map<Cell, byte[]> getWithExpectedNumberOfCells(
+            TableReference tableRef, Set<Cell> cells, long expectedNumberOfPresentCells) {
+        try {
+            return getWithLoader(tableRef, cells, (tableReference, cachedCells, toRead) -> {
+                        long nonEmptyValuesInCache = cachedCells.values().stream()
+                                .filter(value -> value != PtBytes.EMPTY_BYTE_ARRAY)
+                                .count();
+                        long numberOfCellsExpectingValuePostCache =
+                                expectedNumberOfPresentCells - nonEmptyValuesInCache;
+
+                        return Futures.immediateFuture(super.getWithExpectedNumberOfCells(
+                                tableReference, toRead, numberOfCellsExpectingValuePostCache));
```

Comment:
I wanted to add some tests to this class verifying that we call `super...` with less expected values if we have things cached, but couldn't think of a good way to do it... any ideas?

Reply:
Hmm. It is tangential, but if we pass in a mock delegate to the constructor we can verify stuff on it right? Admittedly this kind of tests ForwardingTransaction itself too but I think that's fine